concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -129,6 +129,13 @@ migratePersistentBakerInfoEx StateMigrationParametersP7ToP8{} = migrateReference
         BakerInfoEx av1 ->
         m' (BakerInfoEx av2)
     migrateBakerInfoExV1 BakerInfoExV1{..} = return BakerInfoExV1{_bieIsSuspended = CTrue False, ..}
+migratePersistentBakerInfoEx StateMigrationParametersP8ToP9{} = migrateReference migrateBakerInfoExV1
+  where
+    migrateBakerInfoExV1 ::
+        (AVSupportsDelegation av1, AVSupportsDelegation av2, SupportsValidatorSuspension av1 ~ SupportsValidatorSuspension av2, Monad m') =>
```

Comment:
Given you want to migrate from P8 -> P9, maybe the constraint on validator suspension should read that both protocol support it.

Reply:
Well, the constrains are there, because they are needed, and in this case what is needed is `SupportsValidatorSuspension av1 ~ SupportsValidatorSuspension av2`. 

If we want to be concrete we could just contraint that `av1 ~ 'AccountV4, av2 ~ 'AccountV5` since this is necessarily the case for `StateMigrationParametersP8ToP9` and likewise above. 

So I think we should either keep the generality or make it concrete.

Reply:
To me it would make more sense for the constraint to be more narrow, the same as for delegation. I guess it's a general question. Should we make the constraints as concrete as possible or as general as possible.

Reply:
I realized this case just boils down to `migrateReference (return . coerceBakerInfoExV1)` which we also do in other places in the file.